src/Platforms/devices.h

```diff
@@ -155,37 +172,43 @@ inline void set_appropriate_device_num(int num)
   int deviceCount;
   cudaGetDeviceCount(&deviceCount);
   int num_appropriate=0, device=0;
-  bool set_cuda_device=false;
+  bool set_cuda_device=true;
```

Comment:
Why the logic is changed?

Reply:
The function's scope changed a little by also populating the gpu::device_group_numbers array. This makes it necessary for the loop to run through fully (as opposed to before where it could stop once the device was set) which necessitates  a test if the device still has to be set. Since I like positive tests (if(true) rather than if(!true)) in loops, the variable set_cuda_device changed meaning to "a device still has to be set", rather than "has been set" as before.
Long story short, I was thinking of renaming the variable but the name is logical enough. I can add a comment what the variable does at its declaration.

Reply:
I feel it is more clear to extend the name from set_cuda_device to need_to_set_cuda_device

Reply:
It'll be in the next upload.